### PR TITLE
[ADD] pos_hide_partner_info

### PR DESCRIPTION
**Pull request by victor-champonnois**
Hides phone and address field in pos customer list.

### CODE REVIEW

File: pos_hide_partner_info/static/src/xml/templates.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                <td>
+                    <t t-esc='partner.name' />
+                </td>
+            </tr>
+        </t>
+    </t>
+
+    <t t-extend="ClientDetails">
+        <t t-jquery=".client-details-left" t-operation="replace">
+            <div class='client-details-left' />
```
Comment (legalsylvain) [inline pos_hide_partner_info/static/src/xml/templates.xml]: sale remark as https://github.com/OCA/pos/pull/879

For exemple, this module can conflict with
-  ``pos_partner_firstname`` (``<t t-jquery=".client-details-left" t-operation="prepend">``)
- ``pos_customer_required_fields`` (``<t t-jquery="div.client-details-left" t-operation="append">``)

**Review by legalsylvain — CHANGES_REQUESTED** [review]
thanks for your contribution. one remark inline.
Do you think, it could be great to have the field list configurable ? (not the topic of the current PR, but can be added in the roadmap).

File: pos_hide_partner_info/static/src/xml/templates.xml (hunk 1 above)
Comment (victor-champonnois) [reply pos_hide_partner_info/static/src/xml/templates.xml]: I don't see a way to select [these `client-detail` elements](https://github.com/odoo/odoo/blob/c53081f10befd4f1c98e46a450ed3bc71a6246ed/addons/point_of_sale/static/src/xml/pos.xml#L346)  that have a child with class `client-address-street`. Apparently it's not possible to select the parent of an element with CSS selectors. 
Is it OK if I make the whole [client-details-left](https://github.com/odoo/odoo/blob/c53081f10befd4f1c98e46a450ed3bc71a6246ed/addons/point_of_sale/static/src/xml/pos.xml#L345) invisible, with a warning in Usage.rst ?

Comment (robinkeunen) [reply pos_hide_partner_info/static/src/xml/templates.xml]: @victor-champonnois I don't think that would work. We will actually use both `pos_partner_firstname` and `pos_hide_partner_info` on our production server :/

Comment (victor-champonnois) [reply pos_hide_partner_info/static/src/xml/templates.xml]: OK I found ways to do it without brute force replacing.

File: pos_hide_partner_info/static/src/xml/templates.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates id="template">
+    <t t-extend="ClientListScreenWidget">
+
+        <t t-jquery="th:contains('Address')" t-operation="attributes">
```
Comment (robinkeunen) [inline pos_hide_partner_info/static/src/xml/templates.xml:5]: I don't think it will work with translations unfortunatly :/

**Review by robinkeunen — CHANGES_REQUESTED** [review]

File: pos_hide_partner_info/static/src/xml/templates.xml (hunk 2 above)
Comment (victor-champonnois) [reply pos_hide_partner_info/static/src/xml/templates.xml:5]: It does I think, I tested it on a DB with french translation and it works)

Comment (victor-champonnois) [reply pos_hide_partner_info/static/src/xml/templates.xml:5]: I ask for it to be tested in the functional tests

**Comment by robinkeunen** [comment]
@legalsylvain any time for another review ?

File: pos_hide_partner_info/static/src/xml/templates.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        <t t-jquery="th:contains('Phone')" t-operation="attributes">
+            <attribute name="class">oe_hidden</attribute>
+        </t>
+    </t>
+
+    <t t-extend="ClientLine">
+        <t t-jquery="td:has([t-esc='partner.address'])" t-operation="attributes">
+            <attribute name="class">oe_hidden</attribute>
+        </t>
+        <t t-jquery="td:has([t-esc*='phone'])" t-operation="attributes">
```
Comment (legalsylvain) [inline pos_hide_partner_info/static/src/xml/templates.xml:17]: ``*`` typo ?

Comment (victor-champonnois) [reply pos_hide_partner_info/static/src/xml/templates.xml:17]: No, it means 'contains "phone"'

Comment (victor-champonnois) [reply pos_hide_partner_info/static/src/xml/templates.xml:17]: https://api.jquery.com/category/selectors/

Comment (legalsylvain) [reply pos_hide_partner_info/static/src/xml/templates.xml:17]: oh ! thanks !

**Review by legalsylvain — APPROVED** [review]

**Review by robinkeunen — APPROVED** [review]

**Pull request merged**